include/matx/transforms/conv.h

```diff
@@ -98,53 +98,30 @@ inline void matxDirectConv1DInternal(OutputType &o, const InType &i,
 }
 
 
-template <typename OutputType, typename InType, typename FilterType>
-void matxDirectConv2DInternal(OutputType &o, InType &i,
-                              FilterType &filter, matxConvCorrMode_t mode,
+template <typename OutputType, typename In1Type, typename In2Type>
+void matxDirectConv2DInternal(OutputType &o, In1Type &in1,
+                              In2Type &in2, matxConvCorrMode_t mode,
                               cudaStream_t stream)
 {
-  MATX_STATIC_ASSERT(OutputType::Rank() == InType::Rank(), matxInvalidDim);
-  MATX_STATIC_ASSERT(FilterType::Rank() == 2, matxInvalidDim);
-
-  using filter_input_t = typename FilterType::scalar_type;
-  auto shmsize = sizeof(filter_input_t) * filter.Size(0) * filter.Size(1);
+  MATX_NVTX_START("", matx::MATX_NVTX_LOG_INTERNAL)
+  
+  MATX_STATIC_ASSERT(OutputType::Rank() == In1Type::Rank(), matxInvalidDim);
+  MATX_STATIC_ASSERT(OutputType::Rank() == In2Type::Rank(), matxInvalidDim);
```

Comment:
eventually it would be good to remove this to allow batching a filter without cloning